### PR TITLE
[SPARK-5698] Do not let user request negative # of executors

### PR DESCRIPTION
**Pull request by andrewor14**
Otherwise we might crash the ApplicationMaster. Why? Please see https://issues.apache.org/jira/browse/SPARK-5698.

@sryza I believe this is also relevant in your patch #4168.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 1]
```diff
@@ -314,6 +314,11 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
    * Return whether the request is acknowledged.
    */
   final override def requestExecutors(numAdditionalExecutors: Int): Boolean = synchronized {
+    if (numAdditionalExecutors < 0) {
```
Comment (sryza) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: This seems like the kind of situation where we should throw an `IllegalArgumentException`.

Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: that's fair

**Comment by andrewor14** [comment]
Alright I'm merging this into master 1.3 thanks.

**Comment by sryza** [comment]
LGTM

**Pull request closed without merging**